src/core/timer.c

```diff
@@ -391,12 +393,16 @@ static void timer_enter_waiting(Timer *t, bool time_change) {
                 if (v->base == TIMER_CALENDAR) {
                         usec_t b, rebased;
 
-                        /* If we know the last time this was
-                         * triggered, schedule the job based relative
-                         * to that. If we don't, just start from
-                         * the activation time. */
+                        /* If AlwaysRealtime is enabled, schedule
+                         * the job based on the last time the trigger unit
+                         * entered inactivity. Otherwise, if we know the
+                         * last time this was triggered, schedule the job
+                         * based relative to that. If we don't, just start
+                         * from the activation time or realtime. */
 
-                        if (dual_timestamp_is_set(&t->last_trigger))
+                        if (t->always_realtime && dual_timestamp_is_set(&trigger->inactive_enter_timestamp))
+                                b = trigger->inactive_enter_timestamp.realtime;
+                        else if (!t->always_realtime && dual_timestamp_is_set(&t->last_trigger))
```

Comment:
i think in case of always_realtime it's probably better to take the MAX of last_trigger and inactive_enter_timestamp, simply because CLOCK_REALTIME potentially jumps forward and backward.

Reply:
I'm thinking to remove the condition `!t->always_realtime`
What happens if `always_realtime` is set and `t->last_trigger` has no realtime set?
It goes to the last else?
Is this the intended behaviour?